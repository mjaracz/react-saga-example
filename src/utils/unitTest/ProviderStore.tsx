import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'

export const ProviderStore: FC = ({ children }) => {
  return (
    <Provider store={store}>
      { children }
    </Provider>
  )
}
