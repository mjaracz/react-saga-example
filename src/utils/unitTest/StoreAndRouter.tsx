import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { store } from '../../redux/store'

export const StoreAndRouter: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        { children }
      </BrowserRouter>
    </Provider>
  )
}
