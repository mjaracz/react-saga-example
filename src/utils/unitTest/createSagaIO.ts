import { stdChannel } from 'redux-saga'
import { EventEmitter } from 'events'


export const createSagaIO = (emitter: EventEmitter) => {
  const channel = stdChannel()
  emitter.on('action', channel.put)

  return {
    channel,
    dispatch: action => emitter.emit('action', action),
    getState: () => ({})
  }
}
