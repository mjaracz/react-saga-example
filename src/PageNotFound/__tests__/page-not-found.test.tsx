import React from 'react'
import { render } from '@testing-library/react'
import { PageNotFound } from '../index'
import { StoreAndRouter } from '../../utils/unitTest/StoreAndRouter'

describe('<PageNotFound />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <StoreAndRouter>
        <PageNotFound/>
      </StoreAndRouter>
    )
    expect(baseElement).toMatchSnapshot()
  })
})
