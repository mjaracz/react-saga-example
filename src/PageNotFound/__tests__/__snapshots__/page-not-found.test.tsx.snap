// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="pageNotFound"
    >
      <h1
        class="pageNotFound__title"
      >
        Page Not Found
      </h1>
      <a
        class="pageNotFound__link"
        href="/"
      >
        Go to home page
      </a>
    </div>
  </div>
</body>
`;
